src/region/RegionCreator.tsx

```diff
@@ -172,13 +178,13 @@ export default function RegionCreator({ canDraw, className, onStart, onStop }: P
     const renderRect = (): void => {
         const { current: isDirty } = regionDirtyRef;
         const { current: regionRect } = regionRectRef;
-        const { height = 0, width = 0, x = 0, y = 0 } = getShape() || {};
+        const shape = getShape();
 
-        if (regionRect && isDirty) {
-            regionRect.setAttribute('height', `${height}`);
-            regionRect.setAttribute('width', `${width}`);
-            regionRect.setAttribute('x', `${x}`);
-            regionRect.setAttribute('y', `${y}`);
+        if (regionRect && shape && isDirty) {
+            // Directly set the style attribute
+            Object.assign(regionRect.style, styleShape(shape));
```

Comment:
Do we prefer spread syntax vs Object.assign?

Reply:
We're not able to assign to `Element.style` directly, since it's readonly. This method works in every browser we support.